File: Selenium/Project/Activity9.java
package project;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;

import io.github.bonigarcia.wdm.WebDriverManager;

public class Activity9 {
	
	WebDriver driver;	
	WebDriverWait wait;
	  @BeforeClass
	    public void beforeClass() {
	        // Set up the Firefox driver
	        WebDriverManager.firefoxdriver().setup();
	        driver = new FirefoxDriver();
	     // Initialize wait object
	        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
	        //Open browser
	        driver.get("https://alchemy.hguy.co/lms/");
	    }
	  @Test
	    public void Verify_Lesson_Completed() {
		  //Login to application
		  driver.findElement(By.xpath("//*[contains(text(),'My Account')]")).click();
		  String title = driver.getTitle();
		  System.out.println("Title of the front Page is : " +  title);
		  driver.findElement(By.xpath("//*[contains(text(),'Login')]")).click(); 
		  driver.findElement(By.id("user_login")).sendKeys("root");
		  driver.findElement(By.id("user_pass")).sendKeys("pa$$w0rd");
		  driver.findElement(By.id("wp-submit")).click();
		  
		  //Select the menu item that says “All Courses” and click it.
		  driver.findElement(By.xpath("//*[contains(text(),'All Courses')]")).click();
		  String Page_title = driver.getTitle();
		  System.out.println("Title of the All Courses Page is : " + Page_title);
		  //Select any course and open it.
		  driver.findElement(By.xpath("//article[@id='post-69']/div[2]/p[2]/a[1]")).click();
		  //Select Developing Strategy from course
		  driver.findElement(By.xpath("//div[contains(text(),' Developing Strategy ')]")).click();
		  String Course_title = driver.getTitle();
		  System.out.println("Title of the course is : " + Course_title);  
		  //Select Lesson from Developing Strategy course
		  driver.findElement(By.xpath("//span[contains(text(),'This is the First Topic')]")).click();
		  String Lesson_title = driver.getTitle();
		  System.out.println("Title of the Lesson is : " + Lesson_title);  
		
	  }
	  
	  
	  
	  @AfterClass
	    public void afterClass() {
	        //Close browser
	        driver.close();
	    }

}

File: Selenium/Project/Activity9.py
# Import webdriver from selenium
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.support.wait import WebDriverWait

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
     # Navigate to the URL
    driver.get("https://alchemy.hguy.co/lms/")

    wait = WebDriverWait(driver, 10)

    My_Account = driver.find_element(By.XPATH, "//a[contains(text(),'My Account')]")
    My_Account.click()
    tittle = driver.title
    print("Title of the page : ", tittle)
    driver.find_element(By.XPATH, "//*[contains(text(),'Login')]").click()
    driver.find_element(By.ID, "user_login").send_keys("root")
    driver.find_element(By.ID, "user_pass").send_keys("pa$$w0rd")
    driver.find_element(By.ID, "wp-submit").click()
    #Select the menu item that says “All Courses” and click it.
    driver.find_element(By.XPATH, "//*[contains(text(),'All Courses')]").click()
    Page_tittle = driver.title
    print("Title of the All Courses Page is : ", Page_tittle)
    #Select any course and open it.
    driver.find_element(By.XPATH, "//article[@id='post-69']/div[2]/p[2]/a[1]").click()
    #Select Developing Strategy from course
    driver.find_element(By.XPATH, "//div[contains(text(),' Developing Strategy ')]").click()
    Course_title = driver.title
    print("Title of the All Courses Page is : ", Course_title)
    #Select Lesson from Developing Strategy course
    driver.find_element(By.XPATH, "//span[contains(text(),'This is the First Topic')]").click()
    Lesson_title = driver.title
    print("Title of the Lesson is : ", Lesson_title)
    
